### PR TITLE
Add `TeaserBlock`

### PR DESCRIPTION
**Pull request by SebiVPS**
---

## PR Checklist

-   [x] Link to the respective task if one exists: COM-776 & COM-777
-   [x] Provide screenshots/screencasts if the change contains visual changes

<details>
    <summary>Screenshots/screencasts</summary>
desktop:
Screenshot 2024-07-16 at 16 09 49

tablet:
Screenshot 2024-07-16 at 16 09 59

phone:
Screenshot 2024-07-16 at 16 10 11

</details>

### CODE REVIEW

File: site/src/documents/pages/blocks/TeaserItemBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+);
+
+const Root = styled.div`
+    display: flex;
+    flex: 1 0 80%;
+    justify-content: center;
+
+    ${({ theme }) => theme.breakpoints.sm.mediaQuery} {
+        flex: 1 0 21%;
+    }
```
Comment (jamesricky) [inline site/src/documents/pages/blocks/TeaserItemBlock.tsx:50]: Why are the percent values needed?
Should this not be aligned by the grid?

File: site/src/documents/pages/blocks/TeaserItemBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    display: flex;
+    flex: 1;
+    flex-direction: row;
+    gap: ${({ theme }) => theme.spacing.S300};
+
+    ${({ theme }) => theme.breakpoints.sm.mediaQuery} {
+        flex: unset;
+        gap: ${({ theme }) => theme.spacing.S400};
+        flex-direction: column;
+        max-width: 350px;
```
Comment (jamesricky) [inline site/src/documents/pages/blocks/TeaserItemBlock.tsx:64]: Why is this `max-width` needed?
Should this not be aligned by the grid?

File: site/src/documents/pages/blocks/TeaserItemBlock.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    &:hover {
+        color: ${({ theme }) => theme.palette.primary.dark};
+    }
+`;
+
+const ArrowIcon = styled(SvgUse)`
+    width: 16px;
+    height: 16px;
+    color: inherit;
```
Comment (jamesricky) [inline site/src/documents/pages/blocks/TeaserItemBlock.tsx:95]: Isn't this the default behavior?

File: site/src/documents/pages/blocks/TeaserItemBlock.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+`;
+
+const LinkText = styled.span`
+    font-family: ${({ theme }) => theme.fontFamily};
+    font-size: 16px;
+    font-weight: 700;
+    color: inherit;
```
Comment (jamesricky) [inline site/src/documents/pages/blocks/TeaserItemBlock.tsx:102]: Isn't this the default behavior?

File: site/src/documents/pages/blocks/TeaserItemBlock.tsx (hunk 1 above)
Comment (SebiVPS) [reply site/src/documents/pages/blocks/TeaserItemBlock.tsx:50]: > Why are the percent values needed?

Its flex aligned, what values would you suggest? (1 item for small, 4 items for sm+ breakpoint)

> Should this not be aligned by the grid?

No its not aligned to the grid, its flex aligned

File: site/src/documents/pages/blocks/TeaserItemBlock.tsx (hunk 2 above)
Comment (SebiVPS) [reply site/src/documents/pages/blocks/TeaserItemBlock.tsx:64]: > Why is this max-width needed?

If only one item is in one row, we do not want to use the full width, that would not look good

> Should this not be aligned by the grid?

No its not aligned to the grid, its flex aligned

**Review by jamesricky — APPROVED** [review]

**Review by nsams — APPROVED** [review]

**Comment by johnnyomair** [comment]
Please change the base to `main`.

File: admin/src/documents/pages/blocks/TeaserBlock.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import { HeadingBlock } from "@src/common/blocks/HeadingBlock";
+import { TeaserItemBlock } from "@src/documents/pages/blocks/TeaserItemBlock";
+import { FormattedMessage } from "react-intl";
+
+const TeaserItemsBlock = createListBlock({
+    name: "TeaserItems",
+    displayName: <FormattedMessage id="teaserItemsBlock.displayName" defaultMessage="Teaser items" />,
+    block: TeaserItemBlock,
+    itemName: <FormattedMessage id="teaserItemsBlock.itemName" defaultMessage="teaser item" />,
+    itemsName: <FormattedMessage id="teaserItemsBlock.itemsName" defaultMessage="teaser items" />,
```
Comment (johnnyomair) [inline admin/src/documents/pages/blocks/TeaserBlock.tsx:11]: How about using only "teaser" or "item", but not both?

File: admin/src/documents/pages/blocks/TeaserBlock.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        name: "Teaser",
+        displayName: <FormattedMessage id="teaserBlock.displayName" defaultMessage="Teaser" />,
+        blocks: {
+            heading: {
+                block: HeadingBlock,
+            },
+            items: {
+                block: TeaserItemsBlock,
+                title: <FormattedMessage id="teaserBlock.items" defaultMessage="Teaser Items" />,
```
Comment (johnnyomair) [inline admin/src/documents/pages/blocks/TeaserBlock.tsx:24]: Here as well?

**Review by jamesricky — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

File: site/src/documents/pages/blocks/TeaserItemBlock.tsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+    ${({ theme }) => theme.breakpoints.sm.mediaQuery} {
+        flex: unset;
+        gap: ${({ theme }) => theme.spacing.S400};
+        flex-direction: column;
+    }
+`;
+
+const ContentContainer = styled.div`
+    display: flex;
```
Comment (mariokemetinger) [inline site/src/documents/pages/blocks/TeaserItemBlock.tsx:54]: what is the sense of flex here? I should behave the same way like a standard display block div i think

Comment (SebiVPS) [reply site/src/documents/pages/blocks/TeaserItemBlock.tsx:54]: for the small layout it is needed, flex layouted both with flex:1 for 1 : 1 layout:

Screenshot 2024-07-16 at 11 57 47

**Review by jamesricky — DISMISSED** [review]

File: api/src/documents/pages/blocks/teaser-item.block.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    media: BlockDataInterface;
+
+    @BlockField()
+    title: string;
+
+    @ChildBlock(RichTextBlock)
+    description: BlockDataInterface;
+
+    @ChildBlock(TextLinkBlock)
+    callToAction: BlockDataInterface;
```
Comment (johnnyomair) [inline api/src/documents/pages/blocks/teaser-item.block.ts:28]: Should we still call this call to action, or simply link?

File: api/src/documents/pages/blocks/teaser.block.ts [hunk 9]
```diff
@@ -0,0 +1,4 @@
+import { createListBlock } from "@comet/blocks-api";
+import { TeaserItemBlock } from "@src/documents/pages/blocks/teaser-item.block";
```
Comment (johnnyomair) [inline api/src/documents/pages/blocks/teaser.block.ts:2]: Optional: You should use relative imports for imports in the same module

File: site/src/documents/pages/blocks/TeaserBlock.tsx [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    grid-column: 3 / -3;
+`;
+
+const ItemWrapper = styled.div`
+    display: grid;
+    gap: ${({ theme }) => theme.spacing.D100};
+
+    ${({ theme }) =>
+        css`
+            grid-template-columns: repeat(1, 1fr);
```
Comment (johnnyomair) [inline site/src/documents/pages/blocks/TeaserBlock.tsx:31]: Do you even need a grid for this?

File: site/src/documents/pages/blocks/TeaserItemBlock.tsx [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+
+    &:hover {
+        color: ${({ theme }) => theme.palette.primary.dark};
+    }
+`;
+
+const ArrowIcon = styled(SvgUse)`
+    width: 16px;
+    height: 16px;
+`;
```
Comment (johnnyomair) [inline site/src/documents/pages/blocks/TeaserItemBlock.tsx:98]: The SvgUse components supports the SVG props, so you could pass them as normal DOM attributes: `<SvgUse width="16" height="16" ... />`

File: api/src/documents/pages/blocks/teaser-item.block.ts (hunk 8 above)
Comment (SebiVPS) [reply api/src/documents/pages/blocks/teaser-item.block.ts:28]: In the description stands callToAction, but i also think link fits better 👍  changed it

File: site/src/documents/pages/blocks/TeaserBlock.tsx (hunk 10 above)
Comment (SebiVPS) [reply site/src/documents/pages/blocks/TeaserBlock.tsx]: removed

File: site/src/documents/pages/blocks/TeaserItemBlock.tsx [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+
+const MediaTablet = styled.div`
+    flex: 1;
+    display: none;
+
+    ${({ theme }) => theme.breakpoints.xs.mediaQuery} {
+        display: block;
+    }
+`;
```
Comment (jamesricky) [inline site/src/documents/pages/blocks/TeaserItemBlock.tsx:71]: Maybe name those `MediaMobile` and `MediaDesktop` 🤷🏼‍♂️

**Review by nsams — DISMISSED** [review]

**Review by jamesricky — APPROVED** [review]

File: site/src/documents/pages/blocks/TeaserItemBlock.tsx (hunk 12 above)
Comment (jamesricky) [reply site/src/documents/pages/blocks/TeaserItemBlock.tsx]: Would be ok as well 👍🏼

**Review by jamesricky — DISMISSED** [review]

**Review by johnnyomair — DISMISSED** [review]

**Review by mariokemetinger — DISMISSED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**